src/test/scala/advancepassengerinfo/importer/processor/DqFileProcessorImplTest.scala

```diff
@@ -211,6 +215,65 @@ class DqFileProcessorTest extends TestKit(ActorSystem("MySpec"))
       probe.expectMsg(ManifestCall(jsonFileName, manifest))
       probe.expectMsg(JsonFileCall("c", jsonFileName, successful = true, dateIsSuspicious = false))
       probe.expectMsg(ZipFileCall("c", successful = true))
+
+    }
+
+    "handle exceptions while getting nextFiles and recovery to continue" in {
+      def createManifests(manifest: VoyageManifest): Seq[(String, Try[VoyageManifest])] = Seq(
+        ("manifest1.json", Success(manifest)),
+        ("manifest2.json", Success(manifest)),
+      )
+
+      def createMockFileNamesProvider: FileNames = new FileNames {
+        private var isFirstCall: Boolean = true
+        private val s3Files: String => Future[List[String]] = previous => Future.sequence(List(
+          Future.successful(List("1.zip", "2.zip")),
+          if (isFirstCall) {
+            isFirstCall = false
+            Future.failed(new Exception("next file exception"))
+          } else Future.successful(List("3.zip")),
+        )).map(_.flatten)
+
+        override val nextFiles: (String) => Future[List[String]] = (lastFile: String) => s3Files(lastFile)
+      }
+
+      val mockFileNamesProvider = createMockFileNamesProvider
+      val probe = TestProbe("probe")
+      val mockPersistence = MockPersistence(probe.ref)
+      val manifests = createManifests(manifest)
+      val processor = DqFileProcessorImpl(singleZipMockProvider(manifests), mockPersistence)
+      val dqApiFeed: DqApiFeedImpl = DqApiFeedImpl(mockFileNamesProvider, processor, 100.millis, MockStatsDCollector, LastCheckedState())
+
+      dqApiFeed.processFilesAfter("1.zip").runWith(Sink.seq)
+
+      probe.expectMsg(ManifestCall("manifest1.json", manifest))
+      probe.expectMsg(JsonFileCall("2.zip", "manifest1.json", successful = true, dateIsSuspicious = false))
+      probe.expectMsg(ManifestCall("manifest2.json", manifest))
+      probe.expectMsg(JsonFileCall("2.zip", "manifest2.json", successful = true, dateIsSuspicious = false))
+      probe.expectMsg(ZipFileCall("2.zip", successful = true))
+    }
+
+    "handle exception while persisting and recovery to continue" in {
+      val probe = TestProbe("probe")
+      val mockPersistence = new MockPersistenceWithRecovery(probe.ref)
```

Comment:
Maybe declare this mock inline so we can see that the function throws an exception. At the moment it's not clear that it does as the name doesn't mention it